nvflare/fuel/f3/streaming/byte_streamer.py

```diff
@@ -109,13 +264,10 @@ def send(
         secure=False,
         optional=False,
     ) -> StreamFuture:
-        tx_task = TxTask(channel, topic, target, headers, stream, secure, optional)
+        tx_task = TxTask(self.cell, self.chunk_size, channel, topic, target, headers, stream, secure, optional)
         with self.map_lock:
             self.tx_task_map[tx_task.sid] = tx_task
 
-        future = StreamFuture(tx_task.sid)
-        future.set_size(stream.get_size())
-        tx_task.stream_future = future
         tx_task.task_future = stream_thread_pool.submit(self._transmit_task, tx_task)
```

Comment:
Is the task_future used anywhere else? If not, then no need to keep it with tx_task.
Besides, task_future belongs to tx_task, and shouldn't be set by ByteStreamer object.
This "submit" call could be put into a method of TxTask.

Reply:
Refactored as suggested.